Purchases/Public/RCPurchases.m

```diff
@@ -910,22 +910,28 @@ - (void)handleOfferingsUpdateError:(NSError *)error completion:(RCReceiveOfferin
     CALL_IF_SET_ON_MAIN_THREAD(completion, nil, error);
 }
 
-- (void)receiptData:(void (^ _Nonnull)(NSData * _Nonnull data))completion
-{
+- (void)receiptData:(RCReceiveReceiptDataBlock)completion {
+    [self receiptDataWithForceRefresh:NO completion:completion];
+}
+
+- (void)receiptDataWithForceRefresh:(BOOL)forceRefresh completion:(RCReceiveReceiptDataBlock)completion  {
     NSData *receiptData = [self.receiptFetcher receiptData];
-    if (receiptData == nil) {
+    if (receiptData == nil || receiptData.length == 0) {
         RCDebugLog(@"Receipt empty, fetching");
         [self refreshReceipt:completion];
+    } else if (forceRefresh) {
+        RCDebugLog(@"Forced receipt refresh");
+        [self refreshReceipt:completion];
     } else {
```

Comment:
if we're going to refresh the receipt anyway, this should be the first step, that way we don't make unnecessary calls to read the receipt from disk. 

as in: 

```objective-c
if (forceRefresh) { 
    [self refreshReceipt:completion];
    return;
}

// current logic here ...
NSData *receiptData = [self.receiptFetcher receiptData];
    if (receiptData == nil || receiptData.length == 0) {
    ...
```